server/storage/endpoint/config.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+// ConfigStorage defines the storage operations on the config.
```

Comment:
I prefer move the implements to the `endpoint` package with the interface together.  and remove `bask_bankend` packages, such as:

```
-- endpoint.go
type StorageEndpoint struct {
	kv.Base
	encryptionKeyManager *encryptionkm.KeyManager
}

-- config.go
// ConfigStorage defines the storage operations on the config.
type ConfigStorage interface {
	LoadConfig(cfg interface{}) (bool, error)
	SaveConfig(cfg interface{}) error
	LoadAllScheduleConfig() ([]string, []string, error)
	SaveScheduleConfig(scheduleName string, data []byte) error
	RemoveScheduleConfig(scheduleName string) error
}

func (se * StorageEndpoint) LoadConfig(cfg interface{}) (bool, error) {...}
func (se * StorageEndpoint) SaveConfig(cfg interface{}) error

-- etcd_backend.go
type etcdBackendStorage struct {
	*endpoint.StorageEndpoints
}

// newEtcdBackendStorage is used to create a new etcd backend.
func newEtcdBackendStroage(client *clientv3.Client, rootPath string) *etcdBackend {
	return &etcdBackendStorage{endpoint.NewStorageEndpoint(kv.NewEtcdKVBase(client, rootPath), nil)}
}

....

```
cc @rleungx

Reply:
Maybe put it into an individual file instead of `config.go`

Reply:
BTW, we can use the package name to simplify the naming, like `config.Storage`